### ISSUE TITLE
[Bug]: Modal close button uses wrong title

### ISSUE THREAD

**Issue by huygur** [carbon-design-system/carbon#14323]
### Package

@carbon/react, @carbon/elements

### Browser

Chrome

### Package version

"@carbon/elements": "11.26.0", "@carbon/react": "1.34.0",

### React version

v18.2.0

### Description

Modal close button uses the wrong title even though the closeButtonLabel is set:

![Screenshot 2023-07-27 at 15 16 15](https://github.com/carbon-design-system/carbon/assets/45518829/0d716e73-9e97-494c-b3ea-9117906326b7)

Issue can be reproduced in Storybook: https://react.carbondesignsystem.com/?path=/story/components-modal--playground

According to this code, title gets its value from ariaLabel. Shouldn't it get it from closeButtonLabel? https://github.com/carbon-design-system/carbon/blob/main/packages/react/src/components/Modal/Modal.js#L210

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-modal--playground

### Steps to reproduce

Hover on the close button in Modal component.

### Suggested Severity

None

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tay1orjones** [comment]
@huygur Yep, great catch. `title` should have the same value as `aria-label`

**Comment by SujithPriyamRajan** [comment]
> @huygur Yep, great catch. `title` should have the same value as `aria-label`

## How would it work, @tay1orjones? 
If my `aria-label` is too long, it could affect the title of the close button. Even using `closeButtonLabel` doesn't seem to override the value properly & it doesn't work too.
 Should I consider using an icon inside the modal?
 However, in that case, I can't remove the close icon as I would do with custom inline notifications using `hideCloseButton` Is there an option for this?
